cpp/include/cuvs/neighbors/brute_force.hpp

```diff
@@ -42,34 +45,108 @@ struct index : cuvs::neighbors::ann::index {
   index& operator=(const index&) = delete;
   index& operator=(index&&)      = default;
   ~index()                       = default;
-  index(void* raft_index);
+
+  /** Construct a brute force index from dataset
+   *
+   * Constructs a brute force index from a dataset. This lets us precompute norms for
+   * the dataset, providing a speed benefit over doing this at query time.
+
+   * If the dataset is already in GPU memory, then this class stores a non-owning reference to
+   * the dataset. If the dataset is in host memory, it will be copied to the device and the
+   * index will own the device memory.
+   */
+  template <typename data_accessor>
```

Comment:
I believe this is is happening in brute_force.cu on line 47 (https://github.com/rapidsai/cuvs/pull/118/files#diff-5e5baf58acbb66d86d7b6a3c967c06cfd0b57baabc81cfa61c49da0bb20b5d3bL47)

Reply:
I've removed this `data_accessor` template - and changed to having seperate constructors for host/device datasets , with the implementation in the /src/ directory - since I think its a bit cleaner like this